lib/rules/block-attributes-order.js

```diff
@@ -0,0 +1,320 @@
+/**
+ * @fileoverview enforce ordering of block attributes
+ * @author Wenlu Wang
+ */
+'use strict'
+const utils = require('../utils')
+const { parseSelector } = require('../utils/selector')
+
+/**
+ * @enum {string}
+ */
+const WELL_KNOWN_TEMPLATE_ATTRS = {
+  functional: 'functional',
+  lang: 'lang',
+  src: 'src'
+}
+
+/**
+ * @enum {string}
+ */
+const WELL_KNOWN_SCRIPT_ATTRS = {
+  lang: 'lang',
+  setup: 'setup',
+  src: 'src'
+}
+
+/**
+ * @enum {string}
+ */
+const WELL_KNOWN_STYLE_ATTRS = {
+  scoped: 'scoped',
+  module: 'module',
+  lang: 'lang',
+  src: 'src'
+}
+
+/**
+ * @typedef UserOrderOption
+ * @property { string } element
+ * @property { (string | string[])[] } order
+ */
+
+/**
+ * @typedef {import('../utils/selector').VElementSelector} VElementSelector
+ */
+
+/**
+ * @typedef OrderOption
+ * @property {string} element
+ * @property {(string | string[])[]} order
+ * @property {(RegExp | RegExp[])[]} regexOrder
+ * @property {VElementSelector} selector
+ */
+
+/**
+ * @type {readonly UserOrderOption[]}
+ */
+const defaultOptions = Object.freeze([
+  {
+    element: 'style',
+    order: [
+      WELL_KNOWN_STYLE_ATTRS.module,
+      WELL_KNOWN_STYLE_ATTRS.scoped,
+      WELL_KNOWN_STYLE_ATTRS.lang,
+      WELL_KNOWN_STYLE_ATTRS.src
+    ]
+  },
```

Comment:
I think you can inline them here; since the constants are only used here. I.e.:

```ts
  {
    element: 'style',
    order: ['module', 'scoped', 'lang', 'src']
  },
```